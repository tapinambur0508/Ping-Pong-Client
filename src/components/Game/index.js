import React from 'react';

import Ball from './Ball';
import Court from './Court';
import MyPaddle from './MyPaddle';
import EnemyPaddle from './EnemyPaddle';

class Game extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      width: window.innerWidth,
      height: window.innerHeight
    }
  }

  componentDidMount() {
    window.addEventListener('resize', () => {
      this.setState({
        width: window.innerWidth,
        height: window.innerHeight
      });
    });
  }

  render() {
    return (
        <Court
          width={this.state.width}
          height={this.state.height}>
          <MyPaddle
            windowWidth={this.state.width}
            windowHeight={this.state.height} />
          <Ball
            windowWidth={this.state.width}
            windowHeight={this.state.height} />
          <EnemyPaddle
            windowWidth={this.state.width}
            windowHeight={this.state.height} />
        </Court>
    );
  }
}

export default Game;
