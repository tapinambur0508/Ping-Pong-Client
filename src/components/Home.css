#home .wrapper {
  max-width: 400px;
  margin: 50px auto;
  padding: 30px;
  border-radius: 10px;
  background: #ffffff;
  box-shadow: 0px 2px 2px rgba(0, 0, 0, 0.3);
}

#home .wrapper .heading {
  text-align: center;
  font-family: 'Poppins', sans-serif;
}

#home .wrapper nav {
  margin: 10px 0;
}

#home .wrapper nav a.btn {
  border-radius: 10px;
}