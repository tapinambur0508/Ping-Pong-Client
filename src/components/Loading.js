import React from 'react';

export default ({ error }) => {
  if (error) {
    return <div>Error</div>
  } else {
    return <div>Loading...</div>;
  }
}
