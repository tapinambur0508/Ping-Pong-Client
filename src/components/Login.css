@import url('https://fonts.googleapis.com/css?family=Poppins');
#login .wrapper {
  max-width: 400px;
  margin: 50px auto;
  padding: 30px;
  border-radius: 10px;
  background: #ffffff;
  box-shadow: 0px 2px 2px rgba(0, 0, 0, 0.3);
}

#login .wrapper .heading {
  text-align: center;
  font-family: 'Poppins', sans-serif;
}

#login .wrapper button.kep-login-facebook {
  margin: 10px 0;
  border-radius: 50px;
  text-indent: 10px;
  border: none;
  width: 100%;
}