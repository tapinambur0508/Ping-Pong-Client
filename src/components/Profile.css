#profile .card {
  max-width: 350px;
  margin: 50px auto;
  background: #ffffff;
  box-shadow: 0px 2px 2px rgba(0, 0, 0, 0.3);
}

#profile .card .unit-stats {
  margin: 0;
  color: #ffffff;
  font-weight: 700;
  background: #eda54e;
}

#profile .card .unit-stats .item {
  padding: 1.2rem 1rem;
}

#profile .card .unit-stats .item:nth-child(2) {
  border-left: 1px solid #bc7c2f;
  border-right: 1px solid #bc7c2f;
}

#profile .card .unit-stats .item .stat {
  font-size: 1.5rem;
  margin-bottom: 8px;
}

#profile .card .unit-stats .item .stat-value {
  text-transform: uppercase;
  font-weight: 400;
  font-size: 0.8rem;
}