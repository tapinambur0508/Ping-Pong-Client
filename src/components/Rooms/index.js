import React from 'react';
import axios from 'axios';
import NavLink from 'react-router-dom/NavLink';
import io from 'socket.io-client';

import AuthService from '../../services/auth';

import CreateRoom from './CreateRoom';
import RoomCard from './RoomCard';

import './main.css';

export class Rooms extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      gameRooms: [],
      isDialogOpen: false,
      user: {}
    }

    this.authService = new AuthService();
  }

  componentDidMount() {
    const token = this.authService.token;
    const socket = io(process.env.REACT_APP_API_HOST);

    this.setState({
      user: this.authService.getProfile()
    });

    axios.get('https://ping-pong-main-server.herokuapp.com/api/game-rooms', {
      headers: {
        'Authorization': `Bearer ${token}`
      }
    })
      .then(({ data }) => {
        this.setState({ gameRooms: data['gameRooms'] });
      })
      .catch(err => console.log(err));

    socket.on('newGameRoomCreated', newRoom => {
      const newGameRooms = [newRoom, ...this.state.gameRooms];

      this.setState({ gameRooms: newGameRooms });
    })

    socket.on('gameRoomDeleted', id => {
      const index = this.state.gameRooms.findIndex(e => e._id === id);

      const newGameRooms = [
        ...this.state.gameRooms.slice(0, index),
        ...this.state.gameRooms.slice(index + 1)
      ];

      this.setState({ gameRooms: newGameRooms });
    });
  }

  openDialog = () => {
    this.setState({
      isDialogOpen: true
    });
  }

  closeDialog = () => {
    this.setState({
      isDialogOpen: false
    });
  }

  render() {
    const roomsList = this.state.gameRooms.map(element => (
      <div key={element._id} className="col-4">
        <RoomCard user={this.state.user} {...element} />
      </div>
    ));

    return (
      <section id="rooms">
        <CreateRoom open={this.state.isDialogOpen} onClose={this.closeDialog} />
        <div className="container">
          <div className="card">
            <div className="card-header">
              <div className="dashhead">
                <div className="dashhead-titles">
                  <h2>
                    Game Rooms
                  </h2>
                </div>
                <div className="dashhead-toolbar">
                  <button className="btn btn-primary" onClick={this.openDialog} name="">
                    <i className="fas fa-plus"></i>
                  </button>
                </div>
              </div>
            </div>
            <div className="card-body">
              <div className="row">
                {roomsList}
              </div>
            </div>
            <div className="card-footer">
              <NavLink to="/" className="btn btn-dark">
                <i className="fas fa-arrow-left mr-2"></i>Back
              </NavLink>
            </div>
          </div>
        </div>
      </section>
    );
  }
}

export default Rooms;
