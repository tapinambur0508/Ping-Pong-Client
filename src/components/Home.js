import React from 'react';
import NavLink from 'react-router-dom/NavLink';

import './Home.css';

export default () => (
  <section id="home">
    <div className="container">
      <div className="wrapper">
        <h1 className="heading">Ping-Pong</h1>
        <nav>
          <NavLink to="/start-game" className="btn btn-success btn-block">
            <i className="fas fa-play mr-2"></i>Start Game
            </NavLink>
          <NavLink to="/profile" className="btn btn-info btn-block">
            <i class="far fa-user-circle mr-2"></i>Profile
          </NavLink>
        </nav>
      </div>
    </div>
  </section>
);
