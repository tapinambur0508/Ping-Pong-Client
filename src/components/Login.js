import React from 'react';
import FacebookLogin from 'react-facebook-login';
import withRouter from 'react-router-dom/withRouter';
import axios from 'axios';

import AuthService from '../services/auth';

import config from '../config.json';

import './Login.css';

const Login = props => {
  const facebookResponse = response => {
    axios({
      method: 'POST',
      url: 'https://ping-pong-main-server.herokuapp.com/api/oauth/facebook',
      data: {
        access_token: response.accessToken
      },
      headers: {
        'Content-Type': 'application/json'
      }
    })
      .then(({ data }) => {
        const authService = new AuthService();

        authService.token = data['access_token'];
        props.history.push('/');
      })
      .catch(err => console.log(err));
  };

  return (
    <section id="login">
      <div className="container">
        <div className="wrapper">
          <h1 className="heading">Ping-Pong</h1>
          <FacebookLogin
            appId={config.FACEBOOK_APP_ID}
            autoLoad={false}
            fields="name,email,picture"
            textButton="Continue with Facebook"
            callback={facebookResponse} />
        </div>
      </div>
    </section>
  );
}

export default withRouter(Login);
