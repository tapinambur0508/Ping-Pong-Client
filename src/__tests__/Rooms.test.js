import React from 'react';
import { Rooms } from '../components/Rooms';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { StaticRouter } from 'react-router'

describe('Rooms', () => {
  it('should be defined', () => {
    expect(Rooms).toBeDefined();
  });

  it('Rooms Component matches the snapshot', () => {
    const tree = renderer.create(
      <StaticRouter context={{}}>
        <Rooms />
      </StaticRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should open modal window', () => {
    const wrapper = shallow(<Rooms />);
    const createRoomBtn = wrapper.find('button.btn.btn-primary');

    createRoomBtn.simulate('click');

    const isDialogOpen = wrapper.state().isDialogOpen;

    expect(isDialogOpen).toEqual(true);
  });

  it('should close modal window', () => {
    const wrapper = shallow(<Rooms />);

    const isDialogOpen = wrapper.state().isDialogOpen;

    expect(isDialogOpen).toEqual(false);
  });
})
