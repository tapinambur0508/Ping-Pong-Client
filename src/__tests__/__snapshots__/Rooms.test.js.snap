// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rooms Rooms Component matches the snapshot 1`] = `
<section
  id="rooms"
>
  <div
    className="container"
  >
    <div
      className="card"
    >
      <div
        className="card-header"
      >
        <div
          className="dashhead"
        >
          <div
            className="dashhead-titles"
          >
            <h2>
              Game Rooms
            </h2>
          </div>
          <div
            className="dashhead-toolbar"
          >
            <button
              className="btn btn-primary"
              name=""
              onClick={[Function]}
            >
              <i
                className="fas fa-plus"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        className="card-body"
      >
        <div
          className="row"
        />
      </div>
      <div
        className="card-footer"
      >
        <a
          aria-current="page"
          className="btn btn-dark active"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          <i
            className="fas fa-arrow-left mr-2"
          />
          Back
        </a>
      </div>
    </div>
  </div>
</section>
`;
